plugins/admin/README.adoc

```diff
@@ -0,0 +1,126 @@
+## kn-admin
+
+A kn plugin for Knative cluster management.
+
+`kn hello` prints out a friendly message to the user.
+
+### Description
+
+This kn-admin plugin is designed to help administrators and operators better manage a Knative platform installation with kn CLI.
+The plugin’s main objective is to make administration and operation workflows easier, for instance by making it easy to accomplish
+tasks such as feature flags enablement or disablement with one command, instead of many manual steps like modifying ConfigMaps or yaml files.
+
+### Usage
+
+----
+A plugin of kn client to manage Knative for administrators.
+
+For example:
+kn admin domain set - to set Knative route domain
+kn admin private-registry enable - to enable deployment from the private registry
+
+Usage:
+  admin [command]
+
+Available Commands:
+  domain           Manage route domain
+  help             Help about any command
+  private-registry Manage private-registry
+
+Flags:
+      --config string   config file (default is $HOME/.config/kn/plugins/kn-admin.yaml)
+  -h, --help            help for admin
+  -t, --toggle          Help message for toggle
+
+Use "admin [command] --help" for more information about a command.
+----
+
+#### `kn admin domain`
+
+----
+Set default route domain or route domain for Service with selectors. For example:
+
+kn admin domain set - to set Knative route domain
+
+Usage:
+  admin domain [flags]
+  admin domain [command]
+
+Available Commands:
+  set         set route domain
+
+Flags:
+  -h, --help   help for domain
+
+Global Flags:
+      --config string   config file (default is $HOME/.config/kn/plugins/admin.yaml)
+
+Use "admin domain [command] --help" for more information about a command.
+
+----
+
+#### `kn admin private-registry`
+
+----
+Manage Service deployment from a private registry
+For example:
+
+kn admin private-registry enable \
+  --secret-name=[SECRET_NAME]
+  --docker-server=[PRIVATE_REGISTRY_SERVER_URL] \
+  --docker-email=[PRIVATE_REGISTRY_EMAIL] \
+  --docker-username=[PRIVATE_REGISTRY_USER] \
+  --docker-password=[PRIVATE_REGISTRY_PASSWORD]
+
+Usage:
+  admin private-registry [command]
+
+Available Commands:
+  enable      enable Service deployment from a private registry
```

Comment:
I wonder whether we need another level of command here. (Also the options to this command are not shown here).
Also, I wouldn't call it "private-registry" (as it could be a public registry with password auth, too ?). So maybe "kn admin set-registry --...." or "kn admin registry set" (if you plan a "kn admin registry get", too).

Reply:
Also you then could remove the "--docker" in the options as it is evident for what it is.

Reply:
@rhus
I agree with you. I removed `private` here.
About the command name, I think instead of `enable` or `set`, maybe `add` is more proper here, as user may add not only one registry for its platform, and later I'll add `registry list` and `registry rm` etc.
And `--docker ` is removed as above comment